src/Team-Services-Bot.Api/Dialogs/ApprovalsDialog.cs

```diff
@@ -0,0 +1,204 @@
+        public ApprovalsDialog(IAuthenticationService authenticationService, IVstsService vstsService)
+        {
+            this.authenticationService = authenticationService;
+            this.vstsService = vstsService;
+        }
+
+        /// <summary>
+        /// Gets or sets the account.
+        /// </summary>
+        public string Account { get; set; }
+
+        /// <summary>
+        /// Gets or sets the approvalid.
+        /// </summary>
+        public int ApprovalId { get; set; }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether it is an approval.
+        /// </summary>
+        public bool IsApproval { get; set; }
+
+        /// <summary>
+        /// Gets or sets the profile.
+        /// </summary>
+        public VstsProfile Profile { get; set; }
+
+        /// <summary>
+        /// Gets or sets the Team Project.
+        /// </summary>
+        public string TeamProject { get; set; }
+
+        /// <inheritdoc />
+        public async Task StartAsync(IDialogContext context)
+        {
+            context.Wait(this.ApprovalsAsync);
+
+            await Task.CompletedTask;
+        }
+
+        /// <summary>
+        /// Replies with a list of ApprovalCards.
+        /// </summary>
+        /// <param name="context">The <see cref="IDialogContext"/>.</param>
+        /// <param name="result">The <see cref="IAwaitable{T}"/>.</param>
+        /// <returns>An async <see cref="Task"/>/.</returns>
+        public async Task ApprovalsAsync(IDialogContext context, IAwaitable<IMessageActivity> result)
+        {
+            var activity = await result;
+            var reply = context.MakeMessage();
+
+            this.Account = context.UserData.GetCurrentAccount();
+            this.Profile = context.UserData.GetProfile(this.authenticationService);
+            this.TeamProject = context.UserData.GetCurrentTeamProject();
+
+            if (activity.Text.Equals(CommandMatchApprovals, StringComparison.OrdinalIgnoreCase))
+            {
+                var approvals = await this.vstsService.GetApprovals(this.Account, this.TeamProject, this.Profile);
+                var cards = approvals.Select(a => new ApprovalCard(a)).ToList();
+
+                foreach (var card in cards)
+                {
+                    reply.Attachments.Add(card);
+                }
+
+                reply.AttachmentLayout = AttachmentLayoutTypes.Carousel;
+                await context.PostAsync(reply);
+
+                context.Wait(this.ApproveOrReject);
+            }
+        }
+
+        /// <summary>
+        /// Approves or Rejects an Approval.
+        /// </summary>
+        /// <param name="context">The <see cref="IDialogContext"/>.</param>
+        /// <param name="result">The <see cref="IAwaitable{T}"/>.</param>
+        /// <returns>An async <see cref="Task"/>/.</returns>
+        public async Task ApproveOrReject(IDialogContext context, IAwaitable<IMessageActivity> result)
```

Comment:
Naming best practice says that asynchronous method name should ends with Async. Please rename to ApproveOrRejectAsync().